*** linux-4.8.11/drivers/usb/usbip/stub_tx.c.orig	2017-09-11 10:52:57.000000002 -0700
--- linux-4.8.11/drivers/usb/usbip/stub_tx.c	2017-09-11 10:54:21.000000002 -0700
***************
*** 89,95 ****
  		dev_info(&urb->dev->dev, "device removed?\n");
  		break;
  	default:
! 		dev_info(&urb->dev->dev,
  			 "urb completion with non-zero status %d\n",
  			 urb->status);
  		break;
--- 89,100 ----
  		dev_info(&urb->dev->dev, "device removed?\n");
  		break;
  	default:
! 		/* T.S., 2017/9/11: demoted from dev_info() to dev_dbg()
! 		 * xilinx hw_server (2016.4) produces excessive messages
! 		 * and yet seems to work fine (i.e., error handled or
! 		 * ignored by hw_server). Silence these messages for now...
! 		 */
! 		dev_dbg(&urb->dev->dev,
  			 "urb completion with non-zero status %d\n",
  			 urb->status);
  		break;
