*** linux-4.8.11/arch/arm/boot/dts/zynq-fpga.dtsi.orig	1969-12-31 16:00:00.000000000 -0800
--- linux-4.8.11/arch/arm/boot/dts/zynq-fpga.dtsi	2015-05-19 00:09:20.000000002 -0700
***************
*** 0 ****
--- 1,113 ----
+ /*
+  *  Copyright (C) 2011 - 2014 Xilinx
+  *  Copyright (C) 2012 National Instruments Corp.
+  *
+  * This software is licensed under the terms of the GNU General Public
+  * License version 2, as published by the Free Software Foundation, and
+  * may be copied, distributed, and modified under those terms.
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  */
+ 
+ / {
+ 	amba {
+ 
+ 		ps7-axislv0@43c00000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c00000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 
+ 			/* must subtract 32 from zynq irq ID# PL interrupts are
+ 			 * in two slices: 61...68 and 84..91
+ 			 */
+ 			interrupts = <0 29 4>;
+ 		};
+ 
+ 		ps7-axislv1@43c01000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c01000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 30 4>;
+ 		};
+ 
+ 		ps7-axislv2@43c02000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c02000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 31 4>;
+ 		};
+ 
+ 		ps7-axislv3@43c03000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c03000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 32 4>;
+ 		};
+ 
+ 		ps7-axislv4@43c04000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c04000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 33 4>;
+ 		};
+ 
+ 		ps7-axislv5@43c05000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c05000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 34 4>;
+ 		};
+ 
+ 		ps7-axislv6@43c06000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c06000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 35 4>;
+ 		};
+ 
+ 		ps7-axislv7@43c07000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c07000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 36 4>;
+ 		};
+ 
+ 		ps7-axislv8@43c08000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c08000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 52 4 >;
+ 		};
+ 
+ 		ps7-axislv9@43c09000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c09000 0x1000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 53 4 >;
+ 		};
+ 
+ 		ps7-axislv10@43c20000 {
+ 			compatible = "generic-uio";
+ 			reg = <0x43c20000 0x20000>;
+ 			interrupt-parent = <&intc>;
+ 			interrupts = <0 54 4 >;
+ 		};
+ 	};
+ 
+ };
+ 
+ &clkc {
+ 	fclk-enable      = <0xf>;
+ };
+ 
+ &devcfg {
+ 	clock-names      = "ref_clk", "fclk0", "fclk1", "fclk2", "fclk3";
+ 	clocks           = <&clkc 12 &clkc 15 &clkc 16 &clkc 17 &clkc 18>;
+ 	interrupt-parent = <&intc> ;
+ 	interrupts       = <0 8 4>;
+ };
+ 
+ 
*** linux-4.8.11/arch/arm/boot/dts/zynq-zc706.dts.orig	2015-03-11 12:47:40.557686346 -0700
--- linux-4.8.11/arch/arm/boot/dts/zynq-zc706.dts	2015-03-11 12:47:54.137290648 -0700
***************
*** 13,18 ****
--- 13,19 ----
   */
  /dts-v1/;
  /include/ "zynq-7000.dtsi"
+ /include/ "zynq-fpga.dtsi"
  
  / {
  	model = "Zynq ZC706 Development Board";
*** linux-4.8.11/arch/arm/boot/dts/zynq-zybo.dts.orig	2017-01-06 15:12:54.475418986 -0800
--- linux-4.8.11/arch/arm/boot/dts/zynq-zybo.dts	2017-01-06 15:15:12.289942289 -0800
***************
*** 13,18 ****
--- 13,19 ----
   */
  /dts-v1/;
  /include/ "zynq-7000.dtsi"
+ /include/ "zynq-fpga.dtsi"
  
  / {
  	model = "Zynq ZYBO Development Board";
***************
*** 29,35 ****
  	};
  
  	chosen {
! 		bootargs = "earlycon";
  		stdout-path = "serial0:115200n8";
  	};
  
--- 30,36 ----
  	};
  
  	chosen {
! 		bootargs = "console=ttyPS0,115200 root=/dev/ram earlyprintk";
  		stdout-path = "serial0:115200n8";
  	};
  
***************
*** 67,69 ****
--- 68,74 ----
  	dr_mode = "host";
  	usb-phy = <&usb_phy0>;
  };
+ 
+ &i2c0 {
+ 	status = "okay";
+ };
