*** linux-3.18.11/arch/arm/boot/dts/zynq-7000.dtsi.orig	2015-04-25 17:42:24.583150068 -0700
--- linux-3.18.11/arch/arm/boot/dts/zynq-7000.dtsi	2015-04-25 17:36:24.222688134 -0700
***************
*** 317,323 ****
  
  		watchdog0: watchdog@f8005000 {
  			clocks = <&clkc 45>;
! 			compatible = "xlnx,zynq-wdt-r1p2";
  			device_type = "watchdog";
  			interrupt-parent = <&intc>;
  			interrupts = <0 9 1>;
--- 317,323 ----
  
  		watchdog0: watchdog@f8005000 {
  			clocks = <&clkc 45>;
! 			compatible = "xlnx,zynq-wdt-r1p2", "cdns,wdt-r1p2";
  			device_type = "watchdog";
  			interrupt-parent = <&intc>;
  			interrupts = <0 9 1>;
