*** linux-3.18.11/drivers/clocksource/Kconfig.orig	2015-04-25 21:17:24.663253811 -0700
--- linux-3.18.11/drivers/clocksource/Kconfig	2015-04-25 21:16:08.977730863 -0700
***************
*** 50,56 ****
  	bool
  
  config CADENCE_TTC_TIMER
! 	bool
  
  config CLKSRC_NOMADIK_MTU
  	bool
--- 50,57 ----
  	bool
  
  config CADENCE_TTC_TIMER
! 	bool "Cadence TTC timer"
! 	default n
  
  config CLKSRC_NOMADIK_MTU
  	bool
***************
*** 111,123 ****
  	  hardware anomalies of missing events.
  
  config ARM_GLOBAL_TIMER
! 	bool
  	select CLKSRC_OF if OF
  	help
  	  This options enables support for the ARM global timer unit
  
  config CLKSRC_ARM_GLOBAL_TIMER_SCHED_CLOCK
! 	bool
  	depends on ARM_GLOBAL_TIMER
  	default y
  	help
--- 112,125 ----
  	  hardware anomalies of missing events.
  
  config ARM_GLOBAL_TIMER
! 	bool "ARM global timer"
  	select CLKSRC_OF if OF
+ 	default n
  	help
  	  This options enables support for the ARM global timer unit
  
  config CLKSRC_ARM_GLOBAL_TIMER_SCHED_CLOCK
! 	bool "ARM global timer as sched_clock"
  	depends on ARM_GLOBAL_TIMER
  	default y
  	help
