*** package/e2fsprogs/e2fsprogs.mk.orig	2016-12-29 12:54:33.000000000 -0800
--- package/e2fsprogs/e2fsprogs.mk	2017-01-05 14:46:30.754850234 -0800
***************
*** 31,39 ****
--- 31,45 ----
  	--disable-libuuid \
  	--enable-fsck \
  	--disable-e2initrd-helper \
+ 	--enable-symlink-build \
+ 	--enable-symlink-install \
  	--disable-testio-debug \
  	--disable-rpath
  
+ HOST_E2FSPROGS_CONF_OPTS = \
+ 	--enable-symlink-build \
+ 	--enable-symlink-install
+ 
  ifeq ($(BR2_PACKAGE_E2FSPROGS_FUSE2FS),y)
  E2FSPROGS_CONF_OPTS += --enable-fuse2fs
  E2FSPROGS_DEPENDENCIES += libfuse
