*** package/ncurses/ncurses.mk.orig	2014-09-01 04:20:56.000000000 -0700
--- package/ncurses/ncurses.mk	2014-09-15 09:56:00.000000000 -0700
***************
*** 25,30 ****
--- 25,31 ----
  	--disable-rpath \
  	--disable-rpath-hack \
  	--enable-echo \
+ 	--enable-symlinks\
  	--enable-const \
  	--enable-overwrite \
  	--enable-pc-files \
***************
*** 107,112 ****
--- 108,114 ----
  	--with-shared --without-gpm \
  	--without-manpages \
  	--without-cxx \
+ 	--enable-symlinks \
  	--without-cxx-binding \
  	--without-ada
  
